src/rulesets/Base/sort.jl

```diff
@@ -42,6 +52,12 @@ end
 ##### `sortslices`
 #####
 
+function frule((_, ẋ), ::typeof(sortslices), x::AbstractArray; dims::Integer, kw...)
+    p = sortperm(collect(eachslice(x; dims=dims)); kw...)
+    inds = ntuple(d -> d == dims ? p : (:), ndims(x))
```

Comment:
I vote to kick `dims::Tuple` down the road. It might not be hard to handle, but would need a bit of thought, and some tests...

It will fail with an error now. I don't think Zygote et al. had any hope of getting through `sortslices` before the rule.

Reply:
In fact works fine with OffsetArrays. The generator used for eachslice propagates indices through correctly:
```
julia> x = OffsetMatrix(rand(2,3), 4, 5);

julia> sortslices(x; dims=2)
2×3 OffsetArray(::Matrix{Float64}, 5:6, 6:8) with eltype Float64 with indices 5:6×6:8:
 0.689597  0.805156  0.995562
 0.727762  0.320152  0.924272

julia> rrule(sortslices, x; dims=2)[1]
2×3 OffsetArray(::Matrix{Float64}, 5:6, 6:8) with eltype Float64 with indices 5:6×6:8:
 0.689597  0.805156  0.995562
 0.727762  0.320152  0.924272

julia> collect(eachslice(x; dims=2))
3-element OffsetArray(::Vector{SubArray{Float64, 1, OffsetMatrix{Float64, Matrix{Float64}}, Tuple{Base.Slice{OffsetArrays.IdOffsetRange{Int64, Base.OneTo{Int64}}}, Int64}, true}}, 6:8) with eltype SubArray{Float64, 1, OffsetMatrix{Float64, Matrix{Float64}}, Tuple{Base.Slice{OffsetArrays.IdOffsetRange{Int64, Base.OneTo{Int64}}}, Int64}, true} with indices 6:8:
 [0.9955624968587956, 0.9242722045713299]
 [0.8051558526354236, 0.32015211201093363]
 [0.689597064698794, 0.7277619193702523]
```
I've removed the error message.